packages/terra-dropdown-button/src/_DropdownList.jsx

```diff
@@ -225,6 +226,8 @@ class DropdownList extends React.Component {
         onKeyDown={this.handleKeyDown}
         onKeyUp={this.handleKeyUp}
         role="menu"
+        id={customProps.menuId}
+        aria-labelledby={customProps.buttonId}
```

Comment:
why do we need aria-LabelledBy for `ul` menu ?

Reply:
As suggested by UX team and provided the reference URL to add aria-labelledby for dropdown menu

https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/

Reply:
I had a look on the button navigation and it is not announcing text of button on menu change, it is reading as 'menu label (1 of 4) menu item' please validate at your end as well if required.

Reply:
https://github.com/cerner/terra-core/assets/126655256/f73a9f29-1779-4b3f-bcd1-4405fc11da81

Reply:
you can notice the reported behavior on the screen recording that VO did announced button text as `menu Export, Deafult dropdown` which is not correct for the first dropdown item

Reply:
Updated as discussed.